src/mito/src/engine.rs

```diff
@@ -650,10 +643,17 @@ impl<S: StorageEngine> MitoEngineInner<S> {
             .context(error::AlterTableSnafu { table_name })?;
 
         if let AlterKind::RenameTable { new_table_name } = &req.alter_kind {
-            let mut tables = self.tables.write().unwrap();
-            tables.remove(&table_ref.to_string());
+            // two step lock
+            // should we lock two table names together first?
```

Comment:
Current implementation has an issue, for example:

Thread 1: rename table A to B
Thread 2: rename Table A to C

Both remove A from `tables`, but rename A to B and C. It's not expected.

I think we should check the `self.tables.remove(&table_ref.to_string())` result, it should not be `None`.

If it is `None`, it means that it has already been dropped or renamed.